packages/node-core/src/indexer/storeCache/cacheModel.ts

```diff
@@ -254,14 +269,34 @@ export class CachedModel<
     });
   }
 
-  private getByFieldFromCache(field: keyof T, value: T[keyof T] | T[keyof T][]): T[] {
-    // As getCache always synced the latest record with setCache, from `set` method
-    // We only set getCache here, otherwise we need to search setCache.
-    return Object.entries(this.getCache).map(([, getValue]) => {
-      if ((Array.isArray(value) && includes(value, getValue.data[field])) || isEqual(getValue.data[field], value)) {
-        return getValue.data;
+  private getByFieldFromCache(field: keyof T, value: T[keyof T] | T[keyof T][], findOne?: boolean): T[] {
+    const joinedData: T[] = [];
+    const unifiedIds: string[] = [];
+    if (Object.keys(this.setCache).length !== 0) {
+      Object.entries(this.setCache).map(([, model]) => {
+        if (model.isMatchData(field, value)) {
+          const latestData = model.getLatest().data;
+          unifiedIds.push(latestData.id);
+          joinedData.push(latestData);
+        }
+      });
+      // No need search further
+      if (findOne && joinedData.length !== 0) {
+        return joinedData;
       }
-    });
+    }
+    if (this.getCache.length !== 0) {
+      Object.entries(this.getCache).map(([, getValue]) => {
```

Comment:
How does this work with the cache state? Because every item in the cache is going to be read from.